In [ ]:
# ✅ 1표본 t-검정(좌측 단측검정): "전체 평균 용량이 355ml보다 작다"를 검정
# H0(귀무가설): μ = 355
# H1(대립가설): μ < 355
#
# 주어진 요약 통계:
# n = 200, x̄ = 351, s = 18, α = 0.05

import math
from scipy import stats

# -----------------------------
# 1) 입력값(문제에서 주어진 값)
# -----------------------------
n = 200
x_bar = 351      # 표본평균 (x̄)
s = 18           # 표본표준편차 (s)
mu0 = 355        # 기준 평균 (μ0)
alpha = 0.05     # 유의수준 (α)

# -----------------------------
# 2) t 통계량 계산
# t = (x̄ - μ0) / (s / √n)
# -----------------------------
se = s / math.sqrt(n)          # 표준오차(SE)
t_stat = (x_bar - mu0) / se    # t 통계량
df = n - 1                     # 자유도(df)

# -----------------------------
# 3) p-value (좌측 단측검정)
# p = P(T <= t_stat)
# -----------------------------
p_value = stats.t.cdf(t_stat, df=df)

# -----------------------------
# 4) 임계값(critical value) 방식
# 좌측 단측검정: 하위 α 지점
# -----------------------------
t_crit = stats.t.ppf(alpha, df=df)

# -----------------------------
# 5) 결론
# -----------------------------
reject = (p_value < alpha)

print("=== 1표본 t-검정(좌측 단측검정) 결과 ===")
print(f"n = {n}, df = {df}")
print(f"표본평균 x̄ = {x_bar} ml")
print(f"표본표준편차 s = {s} ml")
print(f"귀무가설 평균 μ0 = {mu0} ml")
print(f"유의수준 α = {alpha}")

print("\n[계산값]")
print(f"표준오차(SE) = s/√n = {se:.4f}")
print(f"t 통계량 = (x̄-μ0)/SE = {t_stat:.4f}")
print(f"p-value (left-tailed) = {p_value:.6f}")
print(f"임계값 t_crit (하위 {alpha*100:.1f}% 지점) = {t_crit:.4f}")

print("\n[결론]")
if reject:
    print("귀무가설(H0) 기각 ✅")
    print("→ 전체적으로 평균 용량이 355ml보다 적게 제공된다고 볼 통계적 근거가 있습니다.")
else:
    print("귀무가설(H0) 기각 못함 ❌")
    print("→ 평균 용량이 355ml보다 작다고 말할 통계적 근거가 부족합니다.")

print("\n[임계값 기준 확인]")
if t_stat <= t_crit:
    print("t_stat <= t_crit 이므로 H0 기각 ✅")
else:
    print("t_stat > t_crit 이므로 H0 기각 못함 ❌")

# -----------------------------
# (선택) 참고: 표본이 매우 크면(df가 커서) t분포는 정규분포에 가까움(근사 가능)
# z 근사(선택):
# z = (x̄ - μ0) / (s / √n)
# p ≈ n


=== 1표본 t-검정(좌측 단측검정) 결과 ===
n = 200, df = 199
표본평균 x̄ = 351 ml
표본표준편차 s = 18 ml
귀무가설 평균 μ0 = 355 ml
유의수준 α = 0.05

[계산값]
표준오차(SE) = s/√n = 1.2728
t 통계량 = (x̄-μ0)/SE = -3.1427
p-value (left-tailed) = 0.000965
임계값 t_crit (하위 5.0% 지점) = -1.6525

[결론]
귀무가설(H0) 기각 ✅
→ 전체적으로 평균 용량이 355ml보다 적게 제공된다고 볼 통계적 근거가 있습니다.

[임계값 기준 확인]
t_stat <= t_crit 이므로 H0 기각 ✅


# **해석**
# 1) 가설 설정
본사가 궁금한 건 “덜 준다(작다)” 방향.

귀무가설(H₀): 평균 용량은 기준과 같다
H0: μ=355H_0:\ \mu = 355H0​: μ=355

대립가설(H₁): 평균 용량은 기준보다 작다
H1: μ<355H_1:\ \mu < 355H1​: μ<355
2) 단측검정/양측검정
👉 단측검정(좌측 단측검정)
이유: 관심이 “355와 다르다”가 아니라, “355보다 작다(덜 준다)” 한 방향뿐이라서.
3) 귀무가설이 기각되었다면? (일상 언어)
200개 지점 측정 결과를 볼 때,
전체적으로 Tall(355ml) 기준보다 평균적으로 적게 제공된다고 볼 통계적으로 유의한 근거가 있다.
⚠️ 표현 주의

❌ “확실히 사기다 / 무조건 덜 준다”



**✅ “통계적으로 유의하다 / 근거가 있다”**

**모표준편차(σ)를 모르는 상황이므로 보통 t-검정을 쓴다.**
**굵은 텍스트**
검정통계량:
t=xˉ−μ0s/nt=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}t=s/n​xˉ−μ0​​
값 대입:

xˉ=351\bar{x}=351xˉ=351, μ0=355\mu_0=355μ0​=355, s=18s=18s=18, n=200n=200n=200
표준오차:
s/n=18/200≈18/14.14≈1.27s/\sqrt{n}=18/\sqrt{200}\approx 18/14.14\approx 1.27s/n​=18/200​≈18/14.14≈1.27
t값:
t=351−3551.27≈−41.27≈−3.15t=\frac{351-355}{1.27}\approx \frac{-4}{1.27}\approx -3.15t=1.27351−355​≈1.27−4​≈−3.15


**유의수준 5% 좌측 단측검정에서 임계값은 대략 -1.65 부근(n이 커서 근사 가능).**

-3.15는 -1.65보다 훨씬 작으므로

✅ 귀무가설 기각

즉, “덜 준다” 쪽으로 통계적으로 강한 신호가 있다는 뜻.
